File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/2112-minimum-difference-between-highest-and-lowest-of-k-scores.c
int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

int minimumDifference(int* nums, int numsSize, int k) {
    if (k < 2) {
        return 0;
    }

    // Step 1: Sort the array
    qsort(nums, numsSize, sizeof(int), compare); // Using quicksort or any sorting algorithm

    // Step 2: Initialize minDifference to a large value
    int minDifference = INT_MAX;

    // Step 3: Sliding window to find minimum difference
    for (int i = 0; i <= numsSize - k; i++) {
        int diff = nums[i + k - 1] - nums[i];  // Difference between the highest and lowest in the window
        if (diff < minDifference) {
            minDifference = diff;
        }
    }

    return minDifference;
}

File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/2112-minimum-difference-between-highest-and-lowest-of-k-scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        if k <= 1:
            return 0  

        nums.sort()  
        mindiff = float('inf') 

       
        for i in range(len(nums) - k + 1):
            diff = nums[i + k - 1] - nums[i] 
            mindiff = min(mindiff, diff) 

        return mindiff
